cart.py

```diff
@@ -0,0 +1,451 @@
+
+        :return: cart position
+        """
+        return self.position
+
+    def set_angle(self, angle):
+        """
+        Sets current pole angle on given value.
+
+        :param angle: angle
+        """
+        self.angle = angle
+
+    def set_position(self, position):
+        """
+        Sets current cart position on given value.
+
+        :param position: position
+        """
+        self.position = position
+
+
+class Environment:
+
+    def __init__(self, system: System, period: float):
+        """
+        Environment represents mathematical model of system with its state-space variables,
+        sampling period, time and cart-pole system.
+        Time is updated every time variables are updated (whenever new force is applied).
+        Variables are rounded (discretized) after every update of variables.
+
+        param system: cart-pole system object
+        param period: sampling period
+        """
+        self.time = 0
+        self.system = system
+        self.period = period
+        self.variables = self._initialize_variables()
+        self._round_values()
+
+    def _initialize_variables(self):
+        """
+        Initializes state-spaces variables of model based on given on system cart and pole.
+
+        :return: list[state_space_variables] in the next order: position, velocity, angle, angle velocity
+        """
+        if self.system.get_angle() > ANGLE_BOUNDARY or self.system.get_angle() < -1 * ANGLE_BOUNDARY:
+            raise Exception("Invalid initial angle given.")
+        else:
+            x1 = self.system.get_position()
+            x2 = 0
+            x3 = self.system.get_angle()
+            x4 = 0
+        return [x1, x2, x3, x4]
+
+    def update_variables(self, f: float):
+        """
+        Updates model state-space variables based on model discretization using Euler 1.
+        When variables are updated, one period has passed,so it's added to time.
+
+        param f: force to be applied on system
+        """
+        self.variables[0] += self.period * self.variables[1]  # x1 = x1 + T * x2
+        self.variables[1] += self.period * self._function(f)  # x2 = x2 + T * Fx(f)
+        self.variables[2] += self.period * self.variables[3]  # x3 = x3 + T * x4
+        self.variables[3] += self.period * self._function(f, position=False)  # x4 = x4 + T * Fo(f)
+        self.time += self.period
+        self._round_values()
+
+    def _function(self, f, position=True):
+        """
+        Function for calculating velocity / angle velocity update.
+        Calculates velocity is parameter position is True and angle velocity if it's False.
+
+        param f: force
+        param position: parameter that indicates which variable is calculated.
+        """
+        m = self.system.get_pole_mass()
+        l = self.system.get_pole_length()
+        M = self.system.get_cart_mass()
+        x3 = self.variables[2]
+        x4 = self.variables[3]
```

Comment:
I dalje nzm sta su ova 2 :(

Reply:
def _initialize_variables(self):
        """
        Initializes state-spaces variables of model based on given on system cart and pole.

        :return: list[state_space_variables] in the next order: position, velocity, angle, angle velocity